### PR TITLE
fix(cb2-7774, cb2-7775): vin and vrms uppercase, added no whitespace directive

### PR DESCRIPTION
**Pull request by leifkemp-bjss**
## Bug: VINs must be capital letters, VRM search should strip spaces on user input before searching

_PR to display amending VINs and VRMs as uppercase, then makes them uppercase in the payloads on submission. Also adds a directive to prevent users from creating whitespaces_
[CB2-7775](https://dvsa.atlassian.net/browse/CB2-7775)
[CB2-7774](https://dvsa.atlassian.net/browse/CB2-7774)

## Checklist

- [ ] Branch is rebased against the latest develop/common
- [ ] Necessary `id` required prepended with `"test-"` have been checked with automation testers and added
- [ ] Code and UI has been tested manually after the additional changes
- [ ] PR title includes the JIRA ticket number
- [ ] Squashed commits contain the JIRA ticket number
- [ ] Link to the PR added to the repo
- [ ] Delete branch after merge

### CODE REVIEW

**Review by shivangidas — DISMISSED** [review]

File: src/app/features/tech-record/components/tech-record-amend-vin/tech-record-amend-vin.component.html [hunk 1]
```diff
@@ -21,11 +21,20 @@ <h2 class="govuk-heading-l">{{ makeAndModel }}</h2>
   </dl>
 
   <form [formGroup]="form">
-    <app-text-input formControlName="vin" name="vin" label="New VIN" [width]="width" (focusout)="isFormValid()"></app-text-input>
+    <app-text-input
+      #vin
```
Comment (mcadenas-bjss) [inline src/app/features/tech-record/components/tech-record-amend-vin/tech-record-amend-vin.component.html:25]: do we still need this?

File: src/app/forms/directives/app-no-space.directive.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { HttpEvent } from '@angular/common/http';
+import { Directive, ElementRef, HostListener } from '@angular/core';
+import { NgControl } from '@angular/forms';
+
+@Directive({
+  selector: '[appNoSpace]'
+})
+export class NoSpaceDirective {
+  constructor() {}
```
Comment (gjulien-bjss) [inline src/app/forms/directives/app-no-space.directive.ts:9]:
```suggestion
```

File: src/app/features/tech-record/components/tech-record-amend-vin/tech-record-amend-vin.component.spec.ts [hunk 3]
```diff
@@ -13,7 +13,7 @@ import { MockStore, provideMockStore } from '@ngrx/store/testing';
 import { TechnicalRecordService } from '@services/technical-record/technical-record.service';
 import { SharedModule } from '@shared/shared.module';
 import { initialAppState } from '@store/index';
-import { updateVinSuccess } from '@store/technical-records';
+import { updateVin, updateVinSuccess } from '@store/technical-records';
```
Comment (gjulien-bjss) [inline src/app/features/tech-record/components/tech-record-amend-vin/tech-record-amend-vin.component.spec.ts:16]:
```suggestion
import { updateVinSuccess } from '@store/technical-records';
```

**Review by mcadenas-bjss — COMMENTED** [review]
a couple of tests for new directive please :)

File: src/app/forms/directives/app-no-whitespace.directive.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import { Directive, HostListener } from '@angular/core';
+import { NgControl } from '@angular/forms';
+
+@Directive({
+  selector: '[formControlName][appNoWhitespace]'
```
Comment (mcadenas-bjss) [inline src/app/forms/directives/app-no-whitespace.directive.ts:5]:
```suggestion
  selector: '[formControlName][appTrimWhiteSpace]'
```

**Review by mcadenas-bjss — COMMENTED** [review]
I don't see spec files for `no-whitespace` or `to-uppercase`. Even if you are testing those in the components using them we need unit tests for the directives themselves.

File: src/app/forms/directives/app-no-space.directive.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+  @HostListener('keydown', ['$event'])
+  public onKeyDown(e: KeyboardEvent): void {
+    console.log(e.key);
```
Comment (tahayazici) [inline src/app/forms/directives/app-no-space.directive.ts:11]: Are the `console.log()`s still needed?

**Review by kpearmanbjss — DISMISSED** [review]

File: src/app/forms/directives/app-no-space.directive.ts [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import { Directive, HostListener } from '@angular/core';
+
+@Directive({
+  selector: '[appNoSpace]'
+})
+export class NoSpaceDirective {
+  @HostListener('keydown', ['$event'])
+  public onKeyDown(e: KeyboardEvent): void {
+    if (e.key === ' ' || e.key === 'Space') {
+      console.log('preventing default');
```
Comment (gjulien-bjss) [inline src/app/forms/directives/app-no-space.directive.ts:10]:
```suggestion
```

**Pull request merged**